From d5459695841525d367b067bacdc0ac756f665b2a Mon Sep 17 00:00:00 2001
From: Daniel van Vugt <daniel.van.vugt@canonical.com>
Date: Fri, 25 Jun 2021 17:09:06 +0800
Subject: [PATCH] magnifier: Avoid painting a desktop that's fully occluded by
 its clone

This significantly reduces the render time for fullscreen zoom mode.

Closes: https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/3305
---
 js/ui/magnifier.js | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/js/ui/magnifier.js b/js/ui/magnifier.js
index d818a75b0d..378bbf5bd1 100644
--- a/js/ui/magnifier.js
+++ b/js/ui/magnifier.js
@@ -851,6 +851,7 @@ var ZoomRegion = class ZoomRegion {
             this._updateMousePosition();
             this._connectSignals();
         } else {
+            Main.uiGroup.set_opacity(255);
             this._disconnectSignals();
             this._destroyActors();
         }
@@ -1397,6 +1398,9 @@ var ZoomRegion = class ZoomRegion {
 
         if (this.isActive() && this._isMouseOverRegion())
             this._magnifier.hideSystemCursor();
+
+        const uiGroupIsOccluded = this.isActive() && this._isFullScreen();
+        Main.uiGroup.set_opacity(uiGroupIsOccluded ? 0 : 255);
     }
 
     _changeROI(params) {
-- 
GitLab

